### PR TITLE
admin fixes

### PR DESCRIPTION
**Pull request by bml2238**
with the more accurate code coverage report, i added some more tests for the admin milestone/task pages plus minor changes to help with the tasks
milestoneID for tasks was also set to possibly be null in the case of unassigned tasks

### CODE REVIEW

File: frontend/crd/src/app/admin/milestone-main-page/milestone-main-page.component.ts [hunk 1]
```diff
@@ -58,10 +58,10 @@ export class MilestoneMainPageComponent implements OnDestroy {
     // https://material.angular.io/components/dialog/overview
     const modalDialog = this.matDialog.open(MilestoneCreateModalComponent, dialogConfig);
 
-    modalDialog.afterClosed().subscribe(result => {
-      //TODO: route to edit page
-      this.ngOnInit();
-    })
+    // modalDialog.afterClosed().subscribe(result => {
```
Comment (jts7382) [inline frontend/crd/src/app/admin/milestone-main-page/milestone-main-page.component.ts:61]: is there a reason to keep this? if not delete this

File: frontend/crd/src/app/admin/task-edit-modal/task-edit-modal.component.ts [hunk 2]
```diff
@@ -43,7 +47,11 @@ export class TaskEditModalComponent implements OnInit {
   }
 
   ngOnInit() { 
-    this.createForm();
+    this.eventService.getEvents().subscribe(events => {
```
Comment (jts7382) [inline frontend/crd/src/app/admin/task-edit-modal/task-edit-modal.component.ts:50]: is this cached or is it making another network call? why not just pass in the event list through the modalData?

File: frontend/crd/src/app/admin/task-main-page/task-main-page.component.ts [hunk 3]
```diff
@@ -21,7 +21,7 @@ export class TaskMainPageComponent implements OnDestroy {
   protected readonly yearLevels = [YearLevel.Freshman, YearLevel.Sophomore, YearLevel.Junior, YearLevel.Senior];
 
   constructor(
-    private taskService: TaskService,
+    public taskService: TaskService,
```
Comment (jts7382) [inline frontend/crd/src/app/admin/task-main-page/task-main-page.component.ts:24]: why is this public now?

**Review by jts7382 — CHANGES_REQUESTED** [review]

**Review by JohnLDavidson — APPROVED** [review]

File: frontend/crd/src/app/admin/task-edit-modal/task-edit-modal.component.ts (hunk 2 above)
Comment (bml2238) [reply frontend/crd/src/app/admin/task-edit-modal/task-edit-modal.component.ts:50]: this is the first time it's called so the task main page doesn't have the list

File: frontend/crd/src/app/admin/task-main-page/task-main-page.component.ts (hunk 3 above)
Comment (bml2238) [reply frontend/crd/src/app/admin/task-main-page/task-main-page.component.ts:24]: testing

**Review by jts7382 — APPROVED** [review]

**Pull request merged**